### PR TITLE
Reorganise Expenditure into Structure incomplete

### PR DESCRIPTION
**Pull request by ShawnTanzc**

### CODE REVIEW

File: src/main/java/seedu/trippie/command/DeleteExpenditureCommand.java [hunk 1]
```diff
@@ -1,7 +1,27 @@
...
+import seedu.trippie.Ui;
+
 import java.util.ArrayList;
 
-public class DeleteExpenditureCommand {
+import static seedu.trippie.ExpenditureList.numberOfItemsTracker;
+
+public class DeleteExpenditureCommand extends Command {
+
+    public DeleteExpenditureCommand(ArrayList<Product> expenditureList, String command) {
```
Comment (ivanderjmw) [inline src/main/java/seedu/trippie/command/DeleteExpenditureCommand.java:12]: I think the variable name `userInput` is better than `command`.

File: src/main/java/seedu/trippie/command/DisplayTotalExpenditureCommand.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import seedu.trippie.ExpenditureList;
+import seedu.trippie.Product;
+import seedu.trippie.Ui;
+
+import java.text.NumberFormat;
+import java.util.ArrayList;
+
+public class DisplayTotalExpenditureCommand extends Command {
+
+    public DisplayTotalExpenditureCommand(ArrayList<Product> expenditureList) throws NumberFormatException {
```
Comment (ivanderjmw) [inline src/main/java/seedu/trippie/command/DisplayTotalExpenditureCommand.java:12]: A better exception name would be `BadFormatException`.

File: src/main/java/seedu/trippie/command/ListExpenditureCommand.java [hunk 3]
```diff
@@ -1,4 +1,27 @@
...
+import seedu.trippie.Ui;
+
+import java.util.ArrayList;
+
+public class ListExpenditureCommand extends Command {
+
+    public ListExpenditureCommand(ArrayList<Product> expenditureList) {
+        if (expenditureList.isEmpty()) {
+            Ui.printLine();
+            System.out.println("There is currently nothing in your expenditure list.");
```
Comment (ivanderjmw) [inline src/main/java/seedu/trippie/command/ListExpenditureCommand.java:14]: This needs to be called from Ui as well, once its implemented.

File: src/main/java/seedu/trippie/command/AddExpenditureCommand.java [hunk 4]
```diff
@@ -1,7 +1,26 @@
...
+
+    public AddExpenditureCommand(ArrayList<Product> expenditureList, String command) {
 
+        String itemName = extractItemName(command);
+        String itemCost = Product.extractItemCost(command);
+        String dayBought = Product.extractDayBought(command);
+        itemEntry = new Product(itemName, itemCost, dayBought);
+        Ui.printLine();
+        System.out.println("Got it! I've added the following item: " + itemEntry.toString());
+        expenditureList.add(itemEntry);
```
Comment (ivanderjmw) [inline src/main/java/seedu/trippie/command/AddExpenditureCommand.java:22]: `userInput` will be a better variable name than `command`.

File: src/main/java/seedu/trippie/command/Command.java [hunk 5]
```diff
@@ -1,4 +1,5 @@
 package seedu.trippie.command;
 
 public class Command {
+
 }
```
Comment (ivanderjmw) [inline src/main/java/seedu/trippie/command/Command.java:5]: This shouldve been an abstract class for all the other commands.

Comment (ivanderjmw) [reply src/main/java/seedu/trippie/command/Command.java:5]: I think creating this class as a parent for everything else will solve the parser error.

**Review by ivanderjmw — APPROVED** [review]
Great job Shawn! Thanks for editing your code to match the structure.

File: src/main/java/seedu/trippie/Product.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        int startIndex = userInput.indexOf("-d") + 2;
+        return userInput.substring(startIndex).replaceAll("[^0-9]","").trim();
+    }
+
+    public static String extractCostFromList(Product spending) {
+        int startIndex = spending.toString().indexOf("- $") + 3;
+        return spending.toString().substring(startIndex);
+    }
+
+    public String getProductItem() {
```
Comment (felixhalim) [inline src/main/java/seedu/trippie/Product.java:40]: It would be better to change to `getProductName();`

**Review by felixhalim — COMMENTED** [review]
Please read my comments

File: src/main/java/seedu/trippie/command/ListExpenditureCommand.java (hunk 3 above)
Comment (felixhalim) [reply src/main/java/seedu/trippie/command/ListExpenditureCommand.java:14]: I think it is okay to put it here since this comment is command-specific.

Comment (felixhalim) [reply src/main/java/seedu/trippie/command/ListExpenditureCommand.java:14]: I think it is okay to call it here since it is command-specific

**Comment by ShawnTanzc** [comment]
will add new PR

File: src/main/java/seedu/trippie/command/AddExpenseCommand.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+
+    @Override
+    public void execute() {
+        List<Expense> expenses = expense.getExpenseList();
```
Comment (ShawnTanzc) [inline src/main/java/seedu/trippie/command/AddExpenseCommand.java:49]: NullExceptionError is not fixed

**Review by felixhalim — APPROVED** [review]
Looks good to merge.

**Pull request merged**